File: Data Structures & Algorithms/contains-duplicate-ii/submission-2.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        var set = new HashSet<Integer>();
        int i = 0;
        for (int j = 0 ; j < nums.length ; j++){
            if(j - i > k){
                set.remove(nums[i]);
                i++;
            }

            if (set.contains(nums[j])){
                return true;
            }

            set.add(nums[j]);
        }

        return false;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        st = set()
        l = 0
        r = 0
        res = 0
        while r < len(s):
            while l < r and s[r] in st:
                st.remove(s[l])
                l += 1

            st.add(s[r])
            res = max(res, r - l + 1)
            r += 1
        
        return res
